ext/NormalizingFlowsCUDAExt.jl

```diff
@@ -0,0 +1,40 @@
+module NormalizingFlowsCUDAExt
+
+using CUDA
+using NormalizingFlows: Random, Distributions
+
+# Make allocation of output array live on GPU.
+function Distributions.rand(
+    rng::CUDA.RNG,
+    s::Distributions.Sampleable{<:Distributions.ArrayLikeVariate,Distributions.Continuous},
+)
+    return @inbounds Distributions.rand!(
+        rng, Distributions.sampler(s), CuArray{float(eltype(s))}(undef, size(s))
+    )
+end
+
+function Distributions.rand(
+    rng::CUDA.RNG,
+    s::Distributions.Sampleable{<:Distributions.ArrayLikeVariate,Distributions.Continuous},
+    n::Int,
+)
+    return @inbounds Distributions.rand!(
+        rng, Distributions.sampler(s), CuArray{float(eltype(s))}(undef, length(s), n)
+    )
+end
```

Comment:
Usage of `length` here will cause some issues, e.g. what if `s` is wrapping a matrix distribution?

Maybe `(undef, size(s)..., n)` will do? But I don't quite recall what is the correct size here; should be somewhere in the Distributions.jl docs.